darts/models/forecasting/varima.py

```diff
@@ -92,15 +102,57 @@ def _fit(
     def _predict(
         self,
         n: int,
+        series: Optional[TimeSeries] = None,
+        historic_future_covariates: Optional[TimeSeries] = None,
         future_covariates: Optional[TimeSeries] = None,
         num_samples: int = 1,
     ) -> TimeSeries:
 
-        super()._predict(n, future_covariates, num_samples)
+        super()._predict(
+            n, series, historic_future_covariates, future_covariates, num_samples
+        )
+
+        if series is not None:
+            self._training_last_values = self._last_values
+            # store new _last_values of the new target series
+            self._last_values = (
+                series.last_values()
+            )  # needed for back-transformation when d=1
+
+            series = self._differentiate_series(series)
+
+            # if the series is differentiated, the new len will be = len - 1, we have to adjust the future covariates
+            if historic_future_covariates and self.d > 0:
+                historic_future_covariates = historic_future_covariates.slice_intersect(
+                    series
+                )
+
+            # updating statsmodels results object state
+            self.model = self.model.apply(
+                series.values(),
+                exog=historic_future_covariates.values()
+                if historic_future_covariates
+                else None,
+            )
+
+        # forecast before restoring the training state
         forecast = self.model.forecast(
```

Comment:
While we are at it, is there a chance we could use something like `simulate()` as we use for ARIMA here, to make VARIMA probabilistic too?